### PR TITLE
Make all paths resolved and absolute

### PR DESCRIPTION
**Pull request by alejoe91**
Unify handling of all paths in kwargs to be `Path(...).absolute()` and add `resolve()` in base

### CODE REVIEW

File: src/spikeinterface/core/base.py [hunk 1]
```diff
@@ -646,10 +646,10 @@ def load(file_path: Union[str, Path], base_folder: Optional[Union[Path, str, boo
         if file_path.is_file():
             # standard case based on a file (json or pickle)
             if str(file_path).endswith(".json"):
-                with open(str(file_path), "r") as f:
+                with open(str(Path(file_path)), "r") as f:
```
Comment (samuelgarcia) [inline src/spikeinterface/core/base.py:649]: Why not `open(file_path), mode="r")` ?

File: src/spikeinterface/core/base.py [hunk 2]
```diff
@@ -646,10 +646,10 @@ def load(file_path: Union[str, Path], base_folder: Optional[Union[Path, str, boo
...
                     d = json.load(f)
             elif str(file_path).endswith(".pkl") or str(file_path).endswith(".pickle"):
-                with open(str(file_path), "rb") as f:
+                with open(str(Path(file_path)), "rb") as f:
```
Comment (samuelgarcia) [inline src/spikeinterface/core/base.py:652]: same comment.

File: src/spikeinterface/extractors/cbin_ibl.py [hunk 3]
```diff
@@ -106,7 +106,10 @@ def __init__(self, folder_path, load_sync_channel=False):
             sample_shifts = get_neuropixels_sample_shifts(self.get_num_channels(), num_channels_per_adc)
             self.set_property("inter_sample_shift", sample_shifts)
 
-        self._kwargs = {"folder_path": str(folder_path.absolute()), "load_sync_channel": load_sync_channel}
+        self._kwargs = {
+            "folder_path": str(Path(folder_path).resolve.absolute()),
```
Comment (samuelgarcia) [inline src/spikeinterface/extractors/cbin_ibl.py:110]:
```suggestion
            "folder_path": str(Path(folder_path).resolve().absolute()),
```

File: src/spikeinterface/extractors/neoextractors/biocam.py [hunk 4]
```diff
@@ -57,11 +59,17 @@ def __init__(
...
+                "file_path": str(Path(file_path).absolute()),
+                "mea_pitch": mea_pitch,
+                "electrode_width": electrode_width,
+            }
+        )
 
     @classmethod
     def map_to_neo_kwargs(cls, file_path):
-        neo_kwargs = {"filename": str(file_path)}
+        neo_kwargs = {"filename": str(Path(file_path).absolute())}
```
Comment (samuelgarcia) [inline src/spikeinterface/extractors/neoextractors/biocam.py:72]: I do not think that all the `map_to_neo_kwargs` were needed.
Only the `self._kwargs` was the problem.
This doesn't hurt but this make the read a bit strange

File: src/spikeinterface/extractors/shybridextractors.py [hunk 5]
```diff
@@ -169,7 +169,7 @@ def __init__(self, file_path, sampling_frequency, delimiter=","):
 
         if Path(file_path).is_file():
             spike_clusters = sbio.SpikeClusters()
-            spike_clusters.fromCSV(str(file_path), None, delimiter=delimiter)
+            spike_clusters.fromCSV(str(Path(file_path).absolute()), None, delimiter=delimiter)
```
Comment (samuelgarcia) [inline src/spikeinterface/extractors/shybridextractors.py:172]: Not needed I think

File: src/spikeinterface/extractors/neoextractors/biocam.py (hunk 4 above)
Comment (alejoe91) [reply src/spikeinterface/extractors/neoextractors/biocam.py:72]: right, it does simplify things

Comment (alejoe91) [reply src/spikeinterface/extractors/neoextractors/biocam.py]: done

File: src/spikeinterface/extractors/shybridextractors.py (hunk 5 above)
Comment (alejoe91) [reply src/spikeinterface/extractors/shybridextractors.py]: done

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**